src/main/java/com/microsoft/dhalion/sensors/MultiSensor.java

```diff
@@ -0,0 +1,113 @@
+import java.util.logging.Logger;
+
+/**
+ * The Multi Sensor consumes metrics from the metrics provider and returns them directly in their current form.
+ * It can be used when the metrics returned by the metrics provider do not require any further manipulation before
+ * being consumed by the detectors. Every time the Multi Sensor is invoked it returns metrics that correspond to the
+ * time window between the timestamp of the previous policy invocation and the current timestamp.
+ * <p>
+ * The Multi Sensor takes as input a collection of metric names, the metrics provider, and a set of components.
+ * It collects the values of the given metrics for these components.
+ */
+public class MultiSensor implements ISensor {
+  private static final Logger LOG = Logger.getLogger(MultiSensor.class.getName());
+  private static final Duration DEFAULT_METRIC_DURATION = Duration.ofSeconds(60);
+
+  private final Collection<String> metricNames;
+  private final Collection<String> components;
+
+  protected ExecutionContext context;
+  protected final MetricsProvider metricsProvider;
+
+  public MultiSensor(Config sysconfig, Collection<String> metricNames, MetricsProvider metricsProvider) {
+    this(sysconfig, metricNames, metricsProvider, null);
+  }
+
+  public MultiSensor(Config policyConf,
+      Collection<String> metricNames,
+      MetricsProvider metricsProvider,
+      Collection<String> components) {
+    this.metricNames = metricNames;
+    this.metricsProvider = metricsProvider;
+
+    if (components != null) {
+      this.components = components;
+    } else {
+      Object result = policyConf.get(Key.CONF_COMPONENT_NAMES.value());
+      if (result != null) {
+        this.components = Arrays.asList(result.toString().split(","));
+      } else {
+        this.components = Collections.emptyList();
+      }
+    }
+  }
+
+  @Override
+  public void initialize(ExecutionContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public Collection<String> getMetricTypes() {
+    return metricNames;
+  }
+
+  /**
+   * Fetches the required metrics
+   *
+   * @return a Collection of Measurements
+   */
+  @Override
+  public Collection<Measurement> fetch() {
+    Instant startTime = context.checkpoint();
+    Duration duration = getDuration();
+    Collection<String> metrics = getMetricTypes();
+    LOG.fine(String.format("Trying to fetch %s @ %s for %s duration", getMetricTypes(), startTime, duration));
+
+    Collection<Measurement> measurements
+        = metricsProvider.getMeasurements(startTime, duration, metrics, getComponents());
+    if (LOG.isLoggable(Level.FINEST)) {
+      measurements.stream().map(Object::toString).forEach(LOG::finest);
+    }
+    return measurements;
+  }
+
+  protected Collection<String> getComponents() {
+    return components;
+  }
+
+  /**
+   * Returns the duration for which the metrics need to be collected
+   *
+   * @return duration in seconds
+   */
+  protected synchronized Duration getDuration() {
```

Comment:
There is a lot of common code between this class and the `BasicSensor`. Would it worth exploring reuse?